Add issue.remove_field_value()

nickbroon opened the pull request:
Complements the existing add_field_value()

nickbroon committed:
feat: Add issue.remove_field_value()

Complements the existing add_field_value()

adehad commented:
@nickbroon thanks for this! Do you think we can add a test (to: tests\resources\test_issue.py) where perhaps we 

1. add the value (perhaps to the label field)
2. check the value is added
3. remove the value
4. check the value is removed
?

nickbroon commented:
> @nickbroon thanks for this! Do you think we can add a test (to: tests\resources\test_issue.py) where perhaps we
> 
> 1. add the value (perhaps to the label field)
> 2. check the value is added
> 3. remove the value
> 4. check the value is removed
>    ?

I added a test that exercises both `remove_field_value()` and `add_field_value()`

nickbroon commented:
The UT failures look unrelated to the changes in the PR.

nickbroon committed:
fix: fields values to add/removed are not always string

for example to add/remove a version the name or id must be used:

issue.add_field_value('fixVersions', {'name': version})`
or
`issue.add_field_value("fixVersions", {"id": fix_version.id})`

Closes: #1724

nickbroon commented:
Rebased to latest main.

I don't have permission to update PR labels. "enhancement" needs changed to "feat"

nickbroon commented:
@adehad I think this is ready for review and consideration of merging.
